ucsmsdk_samples/firmware/ucsfirmware.py

```diff
@@ -913,57 +881,84 @@ def firmware_activate_blade(handle, version, require_user_confirmation=True):
         None
 
     Example:
-        firmware_activate_blade(handle, version="2.2.5b")
+        _firmware_activate(handle, version="2.2.5b")
     """
 
-    blade_bundle = version + "B"
-    # rack_bundle = version + "C"
+    _bundle = version + bundle_type
 
     host_firmware_packs = []
     firmware_running_map = {}
-
-    blades_ = handle.query_classid("ComputeBlade")
-    blades = sorted(blades_, key=lambda blade_: blade_.dn)
-    for blade in blades:
-        blade_dn = blade.dn
-        firmware_running = _get_blade_firmware_running(handle, blade)
+    class_id = None
+
+    if bundle_type == "B":
+        class_id = "ComputeBlade"
+        controller_version = _get_blade_image_firmware_version(
+            handle, bundle_version=version)
+    elif bundle_type == "C":
+        class_id = "ComputeRackUnit"
+        controller_version = _get_rack_image_firmware_version(
+            handle, bundle_version=version)
+
+    servers_ = handle.query_classid(class_id)
+    servers = sorted(servers_, key=lambda server_: server_.dn)
+    server_dns = [server.dn for server in servers]
+
+    for server in servers:
+        server_dn = server.dn
+        firmware_running = _get_server_firmware_running(handle, server)
         if not firmware_running:
-            log.debug("Improper firmware on blade '%s'" % blade_dn)
+            log.debug("Improper firmware on server '%s'" % server_dn)
             continue
-        firmware_running_map[blade_dn] = [firmware_running,
-                                          blade.assigned_to_dn]
-        if firmware_running.version == version:
-            log.debug("Blade ('%s') running software version already at "
-                      "version: '%s'" % (blade_dn, version))
+
+        if firmware_running.version == controller_version:
+            log.debug("Server ('%s') running software version already at "
+                      "version: '%s'" % (server_dn, controller_version))
             continue
         else:
-            log.debug("Blade ('%s') is running at version '%s': Expected '%s'"
-                      % (blade_dn, firmware_running.version, version))
-            assigned_to_dn = blade.assigned_to_dn
+            firmware_running_map[server_dn] = [firmware_running,
```

Comment:
@ragupta-git  Is the above `else` required? I think the else can be done away with so that code can be pulled a level up.